daemon/daemon.go

```diff
@@ -526,11 +559,19 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
 	d.tomb.Kill(nil)
 
 	d.mu.Lock()
-	restartSystem := d.restartSystem != state.RestartUnset
+	restartSystem := (d.restartSystem == state.RestartSystemNow || d.restartSystem == state.RestartSystem)
```

Comment:
can you explain this change? is because of the RestartSocket change?

Reply:
without this change, refreshing snapd leads to a reboot of the machine

Reply:
this change is now more clear, this was also slightly confusing, since we were using `restartSystem` (not `d.restartSystem`) to indicate both a reboot and snapd restart, but the variable was called restartSystem so it was a bit unclear the behavior. This is now more clearer I think